### DISCUSSION TITLE
Get Financial Events discrepancy

### DISCUSSION THREAD

**Discussion by IvanMConnex** [abuzuhri/Amazon-SP-API-CSharp#79] [Q&A]
Hello, how are you doing today? My name is Ivan, and I am with Sync with Connex.

We've been using your SDK for our Amazon integration, and we wanted to get one of our customer's FinancialEvents as part of the buildout, under FinancialService. However, we received errors pertaining to access with ARNs, along the lines of "User: arn:aws:iam::11111:user/sync is not authorized to perform: sts:AssumeRole on resource: amzn1.sellerapps.app.abc123".

We tried to use your SDK and its AuthorizationService to address it, but to no avail, and it was unclear where we need to put in the ARN information and other fields.

Additionally, we actually have our own internal API based off of Amazon's official new API that is working, but not all of the components are operating the way we need them to, hence our interest in your SDK.

I compared the Requests and Responses made by both our company's API and your SDK, and we found that yours obtains FinancialEvents very differently than ours - for one thing, we use GET requests, while you use POST ones.

My 2nd question is essentially why does our GET request work and your POST one doesn't? Which Amazon API version are you using? We're using 2020, but I do recall there was a 2021 version that Amazon was intent on implementing as well. Will the 2020 version soon be deprecated?

We also appear to be sending the same type of Auth calls before trying to get the FinancialEvents using our API and your SDK, so there shouldn't be a problem there.
[AmazonRESTOriginalFinancialEventRq2.txt](https://github.com/abuzuhri/Amazon-SP-API-CSharp/files/7916022/AmazonRESTOriginalFinancialEventRq2.txt)
[AmazonRESTSharpFinancialEventRq2.txt](https://github.com/abuzuhri/Amazon-SP-API-CSharp/files/7916023/AmazonRESTSharpFinancialEventRq2.txt)

Please find our successful GET request and your POST one attached to this message. We look forward to hearing from you. Thank you, and have a good day.

**Comment by abuzuhri** [comment]
Hello @IvanMConnex 

> We've been using your SDK for our Amazon integration, and we wanted to get one of our customer's FinancialEvents as part of the buildout, under FinancialService. However, we received errors pertaining to access with ARNs, along the lines of "User: arn:aws:iam::11111:user/sync is not authorized to perform: sts:AssumeRole on resource: amzn1.sellerapps.app.abc123".


You need to use role not user and the same role should connected to app in amazon [SellingPartnerApiDeveloperGuide](https://github.com/amzn/selling-partner-api-docs/blob/main/guides/en-US/developer-guide/SellingPartnerApiDeveloperGuide.md#step-1-create-an-aws-account)

> I compared the Requests and Responses made by both our company's API and your SDK, and we found that yours obtains FinancialEvents very differently than ours - for one thing, we use GET requests, while you use POST ones.

I can work with you to fix all problem if its exit in my SDK , please send all details to my email and i will check

> My 2nd question is essentially why does our GET request work and your POST one doesn't? Which Amazon API version are you using? We're using 2020, but I do recall there was a 2021 version that Amazon was intent on implementing as well. Will the 2020 version soon be deprecated?

I start API 4 months ago and at the time i only use last version of SP-API endpoint ,
I check the method you want and its should be GET not POST , after i check my SDK i can confirm i use GET also not POST

https://github.com/amzn/selling-partner-api-docs/blob/main/references/finances-api/financesV0.md#listfinancialevents

```Csharp
var list=amazonConnection.Financial.ListFinancialEvents(new ParameterListFinancialEvents()
            {
                PostedAfter=DateTime.UtcNow.AddDays(-15)
            });
```

> We also appear to be sending the same type of Auth calls before trying to get the FinancialEvents using our API and your SDK, so there shouldn't be a problem there.

here sample code with pure c# no dependency at any project i use it for test you can also use it to test the problem 
https://github.com/abuzuhri/Amazon-SP-API-CSharp/tree/main/Others/PureCodeSampleForTest



Please contact me if you want more information

**Comment by abuzuhri** [comment]
I can confirm this code work perfect with me , double check it and let me know if you need any help

```Csharp
var list=amazonConnection.Financial.ListFinancialEvents(new ParameterListFinancialEvents()
            {
                PostedAfter=DateTime.UtcNow.AddDays(-15)
            });
```

**Comment by abuzuhri** [comment]
Also you can try to user this report

```Csharp
ReportManager reportManager = new ReportManager(amazonConnection);
var settlementOrder = reportManager.GetSettlementOrder(90); //GET_V2_SETTLEMENT_REPORT_DATA_FLAT_FILE_V2
```
